wrappers/flutter/lib/src/infra/common/common_fitatu_scanner_preview.dart

```diff
@@ -9,77 +10,174 @@ import '../../scanner_preview_mixin.dart';
 class CommonFitatuScannerPreview extends StatefulWidget {
   const CommonFitatuScannerPreview({
     super.key,
-    required this.onSuccess,
     required this.options,
-    this.onChanged,
+    required this.onResult,
     this.onError,
+    this.onChanged,
+    this.overlayBuilder,
   });
 
-  final ValueChanged<String> onSuccess;
   final ScannerOptions options;
+  final ValueChanged<String?> onResult;
+  final ScannerErrorCallback? onError;
   final VoidCallback? onChanged;
-  final Function(String error)? onError;
+  final PreviewOverlayBuilder? overlayBuilder;
 
   @override
-  State<CommonFitatuScannerPreview> createState() => CommonFitatuScannerPreviewState();
+  State<CommonFitatuScannerPreview> createState() => _CommonFitatuScannerPreviewState();
 }
 
-class CommonFitatuScannerPreviewState extends State<CommonFitatuScannerPreview> with ScannerPreviewMixin {
-  late final MobileScannerController controller;
-  String? scannerError;
+class _CommonFitatuScannerPreviewState extends State<CommonFitatuScannerPreview> with ScannerPreviewMixin, WidgetsBindingObserver {
+  late MobileScannerController controller;
+  MobileScannerArguments? mobileScannerArguments;
+  bool isStarted = false;
+  bool resumeFromBackground = false;
+  int startRetryCount = 0;
 
   @override
   void initState() {
-    controller = MobileScannerController();
-    controller.hasTorchState.addListener(_torchChangeListener);
-    controller.resetZoomScale();
-
+    WidgetsBinding.instance.addObserver(this);
+    controller = MobileScannerController(autoStart: false);
+    controller.torchState.addListener(torchChangeListener);
+    startScanner();
     super.initState();
   }
 
   @override
   void dispose() {
-    controller.hasTorchState.removeListener(_torchChangeListener);
-    controller.dispose();
-    scannerError = null;
-
+    WidgetsBinding.instance.removeObserver(this);
+    controller.torchState.removeListener(torchChangeListener);
+    stopScanner();
```

Comment:
wydaje mi się że to robi już dispose, nie spowoduje to błędów?

Reply:
Co robi dispose?

Reply:
controller.stop()

Reply:
Zrobiłem zmianę. W dispose powinno być controller.dispose, a nie stop